src/components/FavButton/FavButton.component.js

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import { Star } from './FavButton.styles';
+import { BiDislike, BiLike } from 'react-icons/bi';
+
+const FavButton = (item) => {
+  let isFavorite = false;
+  let favorites = [];
+  if (localStorage.getItem('favorites')) {
+    favorites = JSON.parse(localStorage.getItem('favorites'));
+  }
+  if (favorites.find((i) => i.videoId === item.id.videoId)) {
+    isFavorite = true;
+  }
+
+  const addVideo = () => {
+    favorites.push({ videoId: item.id.videoId, info: item });
+    localStorage.setItem('favorites', JSON.stringify(favorites));
+  };
+
+  const removeVideo = () => {
+    favorites = favorites.filter((i) => i.videoId !== item.id.videoId);
+    localStorage.setItem('favorites', JSON.stringify(favorites));
+    console.log(localStorage.getItem('favorites'));
+  };
+
+  return (
+    <>
+      {!isFavorite && (
+        <Star title="Add to favorites">
+          <BiLike onClick={addVideo} />
+        </Star>
+      )}
+
+      {isFavorite && (
```

Comment:
This should be in a ternary operation: isFavorite ? (something if true) : (something if false) . That way the condition is checked only once